azure/services/scalesets/scalesets.go

```diff
@@ -276,6 +278,18 @@ func (s *Service) patchVMSSIfNeeded(ctx context.Context, infraVMSS *azure.VMSS)
 		return nil, errors.Wrap(err, "failed to calculate maxSurge")
 	}
 
+	replicaseManagedByAutoscaler := s.Scope.HasReplicasExternallyManaged(ctx)
+
+	hasCustomDataChanges, err := s.Scope.HasBootstrapDataChanges(ctx)
+	if err != nil {
+		log.V(4).Error(err, "unable to calculate custom data hash, ignoring")
```

Comment:
This error seems important, at least when `replicasManagedByAutoscaler` is true

Maybe another way to approach this:

```
hasCustomDataChanges := false
if s.Scope.HasReplicasExternallyManaged(ctx) {
  hasCustomDataChanges, err := s.Scope.HasBootstrapDataChanges(ctx) { 
    return nil, errors.Wrap(err, "unable to calculate custom data hash")
  }
  if hasCustomDataChanges {
    log.V(4).Info("custom data changed")
  } else {
    log.V(4).Info("custom data unchanged")
  }
}

Reply:
Added change